batch/batch/driver/instance_collection/base.py

```diff
@@ -58,12 +60,17 @@ def register_instance_collection(self, inst_coll: 'InstanceCollection'):
         self.name_inst_coll[inst_coll.name] = inst_coll
 
     def choose_location(
-        self, cores: int, local_ssd_data_disk: bool, data_disk_size_gb: int, preemptible: bool, region: Optional[str]
+        self,
+        cores: int,
+        local_ssd_data_disk: bool,
+        data_disk_size_gb: int,
+        preemptible: bool,
+        regions: List[str],
     ) -> str:
-        if region is None and self.global_live_total_cores_mcpu // 1000 < 1_000:
-            region = self._default_region
+        if self._default_region in regions and self.global_live_total_cores_mcpu // 1000 < 1_000:
```

Comment:
This is saying if we can use the default region because it's in one of the user's selected regions and the cluster size is small, let's just go ahead and use the default region.